src/app/modules/product/product/product.component.html

```diff
@@ -1,6 +1,11 @@
 <div *ngIf="product" class="product">
-  <button *ngIf="role === 'ADMINISTRATOR'" class="product__option-button">
-    Options
+  <button
+    *ngIf="role === 'ADMINISTRATOR' && isAdminPage"
+    class="product__option-button"
+  >
+    <app-product-options
+      (option)="showChosedOption($event)"
+    ></app-product-options>
```

Comment:
Hmm, investigate how you can use the injection token here, also on the client card show the add to cart button on the product card.